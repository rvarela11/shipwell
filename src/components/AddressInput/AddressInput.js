/* eslint-disable camelcase */
// @vendors
import React from 'react';
import PropTypes from 'prop-types';

// @material-ui
import TextField from '@material-ui/core/TextField';

// @utils
import { addUSToAddress } from '../../utils';

const AddressInput = (props) => {
    const {
        geocoded_address,
        isAddressValid,
        label,
        onBlur,
        onChange
    } = props;
    return (
        <TextField
            id={`${label} Address`}
            className='address-input'
            fullWidth
            error={!isAddressValid}
            helperText={(!isAddressValid) ? `Did you mean: ${geocoded_address}` : ''}
            label={label}
            margin="normal"
            onBlur={e => onBlur(addUSToAddress(e.target.value))}
            onChange={e => onChange(e.target.value)}
            type="search"
        />
    );
};

AddressInput.propTypes = {
    geocoded_address: PropTypes.string.isRequired,
    isAddressValid: PropTypes.bool.isRequired,
    label: PropTypes.string.isRequired,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired
};

export default AddressInput;
