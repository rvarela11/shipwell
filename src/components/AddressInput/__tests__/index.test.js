// @vendors
import React from 'react';
import { shallow } from 'enzyme';

// @components
import AddressInput from '../AddressInput';

describe('AddressInput', () => {
    it('renders AddressInput without crashing', () => {
        const wrapper = shallow(
            <AddressInput
                geocoded_address=""
                isAddressValid
                label=""
                onBlur={() => {}}
                onChange={() => {}}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
