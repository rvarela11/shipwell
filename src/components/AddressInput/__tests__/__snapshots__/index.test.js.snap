// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInput renders AddressInput without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressInput
    geocoded_address=""
    isAddressValid={true}
    label=""
    onBlur={[Function]}
    onChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "address-input",
      "error": false,
      "fullWidth": true,
      "helperText": "",
      "id": " Address",
      "label": "",
      "margin": "normal",
      "onBlur": [Function],
      "onChange": [Function],
      "required": false,
      "select": false,
      "type": "search",
      "variant": "standard",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "address-input",
        "error": false,
        "fullWidth": true,
        "helperText": "",
        "id": " Address",
        "label": "",
        "margin": "normal",
        "onBlur": [Function],
        "onChange": [Function],
        "required": false,
        "select": false,
        "type": "search",
        "variant": "standard",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
